src/utils.jl

```diff
@@ -14,6 +14,31 @@ function _error_unsupported_combination(geometry, rule)
     throw(ArgumentError(msg))
 end
 
+################################################################################
+#                           DifferentiationMethod
+################################################################################
+
+# Throw an ArgumentError if Analytical() jacobian not defined for this type
+function _guarantee_analytical(G, diff_method::DifferentiationMethod)
+    throw(ArgumentError("Geometry type $G requires kwarg diff_method = Analytical()"))
+end
+
+_guarantee_analytical(G, ::Analytical) = nothing
+
+# Return whether a geometry type has jacobian methods defined
+_has_analytical(::Type{G}) where {G <: Geometry} = false
+_has_analytical(g::G) where {G <: Geometry} = _has_analytical(G)
+
+# Return the default DifferentiationMethod instance for a particular geometry type
+function _default_method(
```

Comment:
I think we should add the floating point precision as an arg for this method because that would inform a better choice for the step size for `FiniteDifference` rather than always 1e-6.

Reply:
Ah. I hadn’t really put much consideration into how `FP` choice flowed into differential methods. Right now in `jacobian` we have a heuristic that the FP type is the same as the coordinate type, but I’ll have to think about this some more.

Reply:
I don't know that there's a perfect solution for this since `FP` can equally be inferred from the `FiniteDifference` instance or from coordinates, but I've updated the `jacobian` API to better capture the `eltype` of `ts` and enforced that as the `FP` type.

Reply:
I think that is fine, and now for this method we can do something like
```julia
function _default_method(g::Type{G}, T::DataType) where {G <: Geometry}
    return _has_analytical(G) ? Analytical() : FiniteDifference(T)
end

FiniteDifference(T::DataType) = T isa Float32 ? FiniteDifference(1e-3) : FiniteDifference()
```
which should be a better choice when working with Float32.